### DISCUSSION TITLE
Announcing Appwrite 1.0RC1

### DISCUSSION THREAD

**Discussion by Meldiron** [appwrite/appwrite#3769] [Show and tell]
This is the first release candidate for Appwrite 1.0. That's right! We're 🤏 this close to our first stable release. For more information on the changes and installation steps please refer to our [1.0-RC1 Documentation](https://warm-tray-285.notion.site/Appwrite-1-0-0-RC1-a5eab87e115d454db0259f707fc88535).

### DO NOT INSTALL THIS VERSION ON ANY EXISTING INSTANCES OF APPWRITE, THIS IS A RELEASE CANDIDATE AND DOES NOT SUPPORT BACKWARD COMPATIBILITY!!

Here's a summary of everything we've worked on so far 🤩 

## Features
- Added the UI to see the Parent ID of all resources within the UI. [#3653](https://github.com/appwrite/appwrite/pull/3653)
- Added automatic cache cleaning for internal Appwrite services [#3491](https://github.com/appwrite/appwrite/pull/3491)
- Added the ability for Appwrite to handle importing hashed passwords, this can be leveraged to import existing user data from other systems. More information can be found in the document linked above. [#2747](https://github.com/appwrite/appwrite/pull/2747)
- `Users` has now been renamed to `Authentication` within the Appwrite console [#3664](https://github.com/appwrite/appwrite/pull/3664)
- More endpoints were made public (for guests) with proper rate limits [#3741](https://github.com/appwrite/appwrite/pull/3741)
- Added Disqus, Podio, and Etsy OAuth providers [#3526](https://github.com/appwrite/appwrite/pull/3526), [#3488](https://github.com/appwrite/appwrite/pull/3488), [#3522](https://github.com/appwrite/appwrite/pull/3522)
- Function logs now capture stdout [#3656](https://github.com/appwrite/appwrite/pull/3656)
- Added the ability to grant guests write permissions for documents, files and executions [#3727](https://github.com/appwrite/appwrite/pull/3727)

### **Permissions**
- Split `write` into `create`, `update` and `delete`.
- Removed `$read` and `$write` attributes from Documents model
- New `$permissions` array attribute added in Documents model. All permissions go in this array now.
- New syntax to represent various permission types in the `$permissions` array.
- New helper class included in SDKs for Permission and Role
- You can now use **both** collection/bucket level permissions and document/file level permissions. They are now applied with OR logic, meaning access granted at either level is enough.
- Added toggle for file/document level permission.
Here are some examples of the new SDK Permission class

``` javascript
Permission.read() // read a resource
Permission.update() // update a resource
Permission.delete() // delete a resource
Permission.create() // create new resources
Permission.write() // alias for update delete and create
```
Here's  are some examples of the new Role class
``` js
Role.any()
Role.guests()
Role.users()
Role.user([USER_ID])
Role.team([TEAM_ID])
Role.team([TEAM_ID], [ROLE])
```

### **Queries, Indexs, and Pagination**

- New Query helper class in SDK
- Pagination is now a part of queries. You can use query methods to express pagination
- Indexes are now more flexible. Queries provided no longer need to be provided in the exact order as the index.

Here are the new methods:

| Query Method            | SDK Example   | Description                    |
|------------------|---------------------------------------|---------------------------------------------------------------------------------------------------------------------------------|
| equal            | Query.equal("title", ["Iron Man"])    | Returns document if attribute is equal to any value in the provided array. Applies to any indexed attribute.                    |
| notEqual         | Query.notEqual("title", ["Iron Man"]) | Returns document if attribute is not equal to any value in the provided array. Applies to any indexed attribute.                |
| lessThan         | Query.lessThan("score", 10)           | Returns document if attribute is less than the provided value. Applies to any indexed attribute.                                |
| lessThanEqual    | Query.lessThanEqual("score", 10)      | Returns document if attribute is less than or equal to the provided value. Applies to any indexed attribute.                    |
| greaterThan      | Query.greaterThan("score", 10)        | Returns document if attribute is greater than the provided value. Applies to any indexed attribute.                             |
| greaterThanEqual | Query.greaterThanEqual("score", 10)   | Returns document if attribute is greater than or equal to the provided value. Applies to any indexed attribute.                 |
| search           | Query.search("text", "key words")     | Searches string attributes for provided keywords. Applies to any string attribute with a full-text index.                       |
| orderDesc        | Query.orderDesc("attribute")          | Orders results in descending order by attribute. Attribute must be indexed. Pass in an empty string to return in natural order. |
| orderAsc         | Query.orderAsc("attribute")           | Orders results in ascending order by attribute. Attribute must be indexed. Pass in an empty string to return in natural order.  |
| limit            | Query.limit(25)                       | Limits the number of results returned by the query. Used for pagination.                                                        |
| offset           | Query.offset(0)                       | Offset the results returned by skipping some of the results. Used for pagination.                                               |
| cursorAfter      | Query.cursorAfter("62a7...f620")      | Places the cursor after the specified resource ID. Used for pagination.                                                         |
| cursorBefore     | Query.cursorBefore("62a7...a600")     | Places the cursor before the specified resource ID. Used for pagination.                                                        |

### DateTime Attribute
- New DateTime attribute following ISO 8601 in UTC Timezone
- Timestamps returned by all Appwrite models will now also follow DateTime convention

## Bugs
- Fixed an issue where after resetting your password in the Appwrite console, you would not be redirected to the login page. [#3654](https://github.com/appwrite/appwrite/pull/3654)
- Fixed an issue where invalid data could be loaded into the Appwrite console. [#3660](https://github.com/appwrite/appwrite/pull/3660)
- Fixed an issue where users using the MySQL adapter for Appwrite would run into an issue with full text indexes [#154](https://github.com/utopia-php/database/pull/154)
- Fix teams being created with no owners [#3558](https://github.com/appwrite/appwrite/pull/3558)
- Fixed a bug where you could not search users by phone [#3619](https://github.com/appwrite/appwrite/pull/3619)
- Fixed a bug where unaccepted invitations would grant access to projects [#3738](https://github.com/appwrite/appwrite/pull/3738)

## New Contributors

* @iamdineshkumar made their first contribution in https://github.com/appwrite/appwrite/pull/3514
* @ishanvyas22 made their first contribution in https://github.com/appwrite/appwrite/pull/3553
* @iifawzi made their first contribution in https://github.com/appwrite/appwrite/pull/3590
* @Aragur made their first contribution in https://github.com/appwrite/appwrite/pull/3180

---

We are looking forward to your feedback as we get closer to Appwrite 1.0. We would love you to try out this new version and leave your feedback in the comments of this discussion 🙏

**Comment by rodgav** [comment]
how do i test it?

**Reply by christyjacob4** [reply]
@rodgav  head over to our Notion Doc to check out the steps 
https://warm-tray-285.notion.site/Appwrite-1-0-0-RC1-a5eab87e115d454db0259f707fc88535
